gravity.sh

```diff
@@ -640,8 +640,27 @@ gravity_DownloadBlocklistFromUrl() {
     cmd_ext="--resolve $domain:$port:$ip $cmd_ext"
   fi
 
+
+  if [ -f "${piholeDir}/list.${id}.${domain}.etag" ]
+	then 
+	  currentetag=$(<"${piholeDir}/list.${id}.${domain}.etag")
+	else
+	  currentetag=$""
+  fi
+
+  echo -e "\n current Etag: " "$currentetag"
   # shellcheck disable=SC2086
-  httpCode=$(curl -s -L ${compression} ${cmd_ext} ${heisenbergCompensator} -w "%{http_code}" -A "${agent}" "${url}" -o "${patternBuffer}" 2> /dev/null)
+  httpCode=$(curl -s -L ${compression} ${cmd_ext} ${heisenbergCompensator} --trace "$id".log --header "If-None-Match: ${currentetag}" -w "%{http_code}" -A "${agent}" "${url}" -o "${patternBuffer}" 2> /dev/null)
+
+# shellcheck disable=SC2086
+  curl -sI "${url}" -L ${compression} ${cmd_ext} ${heisenbergCompensator} | tr -d "\r" | sed -e "/etag/!d" -e "s/^etag: //" | tr -d "\n" > "${piholeDir}/list.${id}.${domain}.etag"
```

Comment:
Allow ETag header with capital letters and use single quotes for literal strings:
```suggestion
  curl -sI "${url}" -L ${compression} ${cmd_ext} ${heisenbergCompensator} | tr -d '\r' | sed -e '/etag/I!d' -e 's/^etag: //I' | tr -d '\n' > "${piholeDir}/list.${id}.${domain}.etag"
```

Reply:
Still not foolproof :)
`https://urlhaus.abuse.ch/downloads/hostfile/` 
and 
`https://urlhaus.abuse.ch/downloads/text`

get the etag of

```
content-security-policy: default-src 'self' https://fonts.gstatic.com:443 data:; style-src 'self' 'unsafe-inline' https://www.gstatic.com:443 https://fonts.googleapis.com:443; script-src 'self' 'unsafe-inline' 'unsafe-eval' https://www.gstatic.com:443 https://www.google.com/recaptcha/ https://www.googletagmanager.com:443; frame-src https://www.google.com/recaptcha/; img-src 'self' data: https://syndication.twitter.com:443; object-src 'none'"46ba9a-5e325b1fa769d-gzip"
```

this one is also funny
```
 [i] Target: https://gitlab.com/nobodysu/spyware-hosts/raw/master/lists/unity-all.txt
  [i] Status: Pending...
 current Etag:  W/"ee7bf2ecae84e0e39954ef52c279c56a"
 new Etag:       W/"ee7bf2ecae84e0e39954ef52c279c56a"
 statuscode:  200
  [✓] Status: Retrieval successful
  [i] Analyzed 143 domains
  [i] List stayed unchanged
```

Reply:
yes, curl in a recent version would be nice but unfortunately pi hole runs on older versions to so i had to come up with the hack we are trying to make work here.